service/registry/src/main/java/io/helidon/service/registry/Interception.java

```diff
@@ -138,4 +138,51 @@ interface Chain<V> {
      */
     public interface ElementInterceptor extends Interceptor {
     }
+
+    /**
+     * Interceptor around initial entry into the Helidon system.
+     * <p>
+     * The following should be considered entry points:
+     * <ul>
+     *     <li>HTTP Request on the WebServer (includes grpc, initial web-socket request, graphQL)</li>
+     *     <li>Messaging inbound message handled by Helidon component</li>
+     *     <li>WebSocket message</li>
```

Comment:
Intercept every websocket message? gRPC supports message streams, should we intercept every message there as well? Maybe this is too much

Reply:
Yes, I think for certain protocols such as websockets and gRPC, the granularity of interception should be different.